acmw-db-app/pages/forgotPassword.js

```diff
@@ -0,0 +1,89 @@
+import Head from 'next/head'
+import styles from '../styles/Reset.module.css'
+import React, { useState } from 'react';
+  
+class forgotPassword extends React.Component {
+    constructor() {
+    super();
+    this.state = {
+      input: {},
+      errors: {}
+    };
+     
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+     
+  handleChange(event) {
+    let input = this.state.input;
+    input[event.target.name] = event.target.value;
+  
+    this.setState({
+      input
+    });
+  }
+     
+  handleSubmit(event) {
+    event.preventDefault();
+  
+    if(this.validate()){
+        let input = {};
+        input["email"] = "";
+        this.setState({input:input});
+    }
+  }
+  
+  validate(){
+      let input = this.state.input;
+      let errors = {};
+      let isValid = true;
+
+       if ((typeof input["email"] !== "") || (typeof input["email"] !== "undefined") || (typeof input["email"] !== null)) {
+
+        var pattern = new RegExp(/^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i);
+        if (!pattern.test(input["email"] || input["email"] == "")) {
```

Comment:
This should just be `if (!pattern.test(input["email"]) {` since "true" or "false" should not be passed to the test function.